common/testresult.py

```diff
@@ -30,6 +34,9 @@ def append(self, testresult):
         self.total_execution_time = total_execution_time
 
     def serialize(self):
+        for i in self.results:
+            if isinstance(i.exception, Py4JJavaError):
+                i.exception = Exception(str(i.exception))

```

Comment:
This may limit the error info. Perhaps it would be better to use `traceback.format_exception`

Reply:
From the [Py4J documentation](https://www.py4j.org/py4j_java_protocol.html#py4jjavaerror), "str(py4j_java_error) returns the error message and the stack trace available on the Java side (similar to printStackTrace())." It seemed to be the simplest way to capture the original (Java) error message and stack trace.

Here is an example of what the output looks like after serialising/deserialising the exception:

```python
foo = result.test_results.serialize()
bar = TestResults().deserialize(foo)
print(bar.results[0].exception)
```

Result:

```
An error occurred while calling o415._run.
: com.databricks.WorkflowException: com.databricks.NotebookExecutionException: FAILED
	at com.databricks.workflow.WorkflowDriver.run(WorkflowDriver.scala:71)
	at com.databricks.dbutils_v1.impl.NotebookUtilsImpl.run(NotebookUtilsImpl.scala:122)
	at com.databricks.dbutils_v1.impl.NotebookUtilsImpl._run(NotebookUtilsImpl.scala:89)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)
	at py4j.Gateway.invoke(Gateway.java:295)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:251)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.databricks.NotebookExecutionException: FAILED
	at com.databricks.workflow.WorkflowDriver.run0(WorkflowDriver.scala:117)
	at com.databricks.workflow.WorkflowDriver.run(WorkflowDriver.scala:66)
	... 13 more
```